src/ol/View.js

```diff
@@ -1372,7 +1372,7 @@ class View extends BaseObject {
    */
   getConstrainedZoom(targetZoom, opt_direction) {
     const targetRes = this.getResolutionForZoom(targetZoom);
-    return this.getZoomForResolution(this.getConstrainedResolution(targetRes));
+    return this.getZoomForResolution(this.getConstrainedResolution(targetRes, opt_direction));
```

Comment:
I'm unsure how this parameter is supposed to be used. Maybe it should be inverted, since the zoom level increases when the resolution decreases? But it's not tested so not 100% positive on this.

Reply:
And this `opt_direction` param is never used in the library so maybe it should be removed?

Reply:
Depends if it was part of the API in 5.x, and also it makes sense to have such an option for this method... sorry for ruining the fun!